### DISCUSSION TITLE
Does anything like KotlinPoet exist for Swift?

### DISCUSSION THREAD

**Discussion by erawhctim** [square/kotlinpoet#1067] [Q&A] [answered]
KotlinPoet is _fantastic_ and incredibly useful! I'm curious if a similar library exists for Swift? 

Most libraries I can find available today are built with some kind of template system (e.g. [Sourcery](https://github.com/krzysztofzablocki/Sourcery), [SwiftGen](https://github.com/SwiftGen/SwiftGen)/[Stencil](https://github.com/stencilproject/Stencil), Swift lang's [GYB util](https://nshipster.com/swift-gyb/)), but I'm looking for something built on top of the language grammar/constructs that offers more dynamism and customization.

Are you guys aware of anything like that? Do you do any Swift source code generation internally, similar to how you use KotlinPoet?

**Comment by erawhctim** [comment, marked as answer]
EDIT: I may have spoke too soon? 😅  https://github.com/outfoxx/swiftpoet

**Reply by JakeWharton** [reply]
Yes. This is what we use for Swift.

**Comment by ZacSweers** [comment]
If you search for "swiftpoet" there's a few options out there, including one actually written in kotlin

**Reply by erawhctim** [reply]
thanks Zac!
